src/DynamoPackages/PackageLoader.cs

```diff
@@ -320,9 +320,11 @@ public void LoadPackages(IEnumerable<Package> packages)
             // Setting back the DisableRun property back to false, as the package loading is completed.
             EngineController.DisableRun = false;
 
-            var assemblies =
-                enumerable.SelectMany(x => x.EnumerateAssembliesInBinDirectory().Where(y => y.IsNodeLibrary));
-            OnPackagesLoaded(assemblies.Select(x => x.Assembly));
+            var assemblies = packageList
+                .SelectMany(p => p.LoadedAssemblies.Where(y => y.IsNodeLibrary))
+                .Select(a => a.Assembly)
+                .ToList();
```

Comment:
There are basically two things that were fixed:
1. The call to `EnumerateAssembliesInBinDirectory` was replaced by using `LoadedAssemblies`. This avoids re-adding assemblies to `LoadedAssemblies`.
2. The assemblies list is materialized. Before these changes, the call to `EnumerateAssembliesInBinDirectory` was made each time the enumerable was enumerated, which happened once per event handler for `PackageLoaded`. That caused triplication of the libraries for me on latest master.

Reply:
@mmisol you mentioned that `EnumerateAssembliesInBinDirectory` was being called in a couple of other places. Do those need to be replaced with `LoadedAssmeblies` as well?

Reply:
Yes, you are right. What I was seeing in the stack trace was actually this line of code being executed when the enumeration was materialized. This happened in two event handlers of `OnPackagesLoaded`, one in DynamoModel.cs(978) and another in EngineController.cs(521).

Reply:
@mmisol So I assume whichever event handler called second time, with this fix, we will no longer re-adding assemblies to `LoadedAssemblies`. Do we want to create a test by mocking up the situation and check the end `LoadedAssemblies`?

Reply:
Yes, that was part of the issue. The call to `TryLoadPackageIntoLibrary` is the first to call `EnumerateAssembliesInBinDirectory` and after this `LoadedAssemblies` should be used to read them. 

I'll see to add a test. I have to think about what needs to be mocked, hopefully it's not too difficult.